code/game/machinery/shieldgen.dm

```diff
@@ -328,17 +328,13 @@
 /obj/machinery/shieldwallgen/New()
 	power_connection = new(src)
 	power_connection.monitoring_enabled = TRUE
-	. = ..()
+	..()
 
 /obj/machinery/shieldwallgen/Destroy()
-	src.cleanup(1)
```

Comment:
I'm sorry, this is my fault because I wasn't very clear with you. Just delete `src.`, it still needs to call cleanup 4 times to represent each direction. It's better if you use the defines cleanup(NORTH), cleanup(SOUTH).

The entire proc is really badly written

Reply:
Updated this too, restoring those `cleanup()`s.